go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -332,6 +325,180 @@ func deleteEgressFirewallRules(externalID string) error {
 	return deletionErrors
 }
 
+// addNodeForEgressFirewall creates the default allow ACLs as follows:
+// - allow cluster subnet on its own switch
+// - allows service subnet on its own switch
+// - allow node's IP on every nodes' switch
+func (oc *Controller) addNodeForEgressFirewall(node *v1.Node) {
+	matchConditions := []string{}
+	v4ClusterSubnet, v6ClusterSubnet := getClusterSubnets()
+	v4ServiceSubnet, v6ServiceSubnet := getServiceSubnets()
+	if config.Gateway.Mode == config.GatewayModeLocal {
+		matchConditions = append(matchConditions, generateDefaultClusterSubnetAllowMatch(v4ClusterSubnet, v6ClusterSubnet)...)
+		matchConditions = append(matchConditions, generateDefaultServiceSubnetAllowMatch(v4ServiceSubnet, v6ServiceSubnet, v4ClusterSubnet, v6ClusterSubnet)...)
+	}
+	for _, match := range matchConditions {
+		_, stderr, err := util.RunOVNNbctl(
+			"--may-exist",
+			"acl-add",
+			node.Name,
+			"from-lport",
+			types.EgressFirewallDefaultInternalClusterPriority,
+			match,
+			"allow",
+		)
+		if err != nil {
+			klog.Errorf("Unable to create default allow ACL on switch: %s, stderr: %s, err: %v", node.Name, stderr, err)
+		}
+	}
+	matchConditions = oc.generateDefaultNodeAllowMatch(node, v4ClusterSubnet, v6ClusterSubnet)
+	logicalSwitches := oc.getEgressFirewallACLSwitches()
+	for _, logicalSwitch := range logicalSwitches {
+		for _, match := range matchConditions {
+			_, stderr, err := util.RunOVNNbctl(
+				"--may-exist",
+				"acl-add",
+				logicalSwitch,
+				"from-lport",
+				types.EgressFirewallDefaultInternalClusterPriority,
+				match,
+				"allow",
+			)
+			if err != nil {
+				klog.Errorf("Unable to create default allow ACL on switch: %s, stderr: %s, err: %v", logicalSwitch, stderr, err)
+			}
+		}
+	}
+}
+
+// deleteNodeForEgressFirewall deletes the default allow ACLs as follows:
```

Comment:
Yes, I tried the second option you mentioned when implementing this, but I felt it obfuscated the method more than increasing the readability. Given that the arguments to `RunOVNNbctl` are not the same if we perform an add/delete, I think I'll keep it like this for now since I don't think splitting the function in two provides a lot of benefit for such a short function.